Source/Core/CesiumTerrainProvider.js

```diff
@@ -556,6 +622,42 @@ define([
         });
     }
 
+    function findNumberOfLevels(value) {
+        // This finds the number of levels stored in a BVH extension tile.
+        // All levels need to be complete.
+        // See http://mikestoolbox.com/powersum.html
+        for (var i = 0, current = 1; i < 32; ++i, current *= 4) {
+            if (value === ((current-1) / 3)) {
+                return i;
+            }
+        }
+
+        return undefined;
+    }
+
+    function recurseHeights(level, x, y, buffer, index, maxLevel, providerAvailability, layerAvailability) {
+        if (level > maxLevel) {
+            return index;
+        }
+
+        if (!isNaN(buffer[index])) {
+            // Minimum height isn't a Nan, so the tile exists
+            // TODO: Make this more efficient
```

Comment:
We generally don't merge PRs with `TODOs`.  Write an issue or just do it.  Is it a bottleneck in practice?